### PR TITLE
env: setting up the backend environment

### PR DESCRIPTION
**Pull request by zoomKoding**
- backend initial environment setup
- DB connection created
- lint and prettier implemented
- base model class implemented
- user model implemented using base model class
- login, signup functionality implemented

### CODE REVIEW

File: server/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const bcrypt = require('bcrypt')
+const User = require('../models/user')
+
+exports.create = async (req, res) => {
+  const input = req.body
+  const salt = await bcrypt.genSalt(+process.env.SALTROUNDS)
+  input.password = await bcrypt.hash(input.password, salt)
```
Comment (jhaemin) [inline server/controllers/user.js:7]: Why explicit `+`?

Comment (zoomKoding) [reply server/controllers/user.js:7]: Used it to convert the string formatted number!

Comment (zoomKoding) [reply server/controllers/user.js:7]: Here is the reference!
[Unary_plus](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus)

Comment (jhaemin) [reply server/controllers/user.js:7]: So `+process.env.SALTROUNDS` becomes a number since `.env` variables are resolved as string. Great.

**Pull request merged**